atlasdb-ete-test-utils/src/main/java/com/palantir/atlasdb/todo/generated/SnapshotsMetadataCleanupTask.java

```diff
@@ -25,14 +28,29 @@ public SnapshotsMetadataCleanupTask(Namespace namespace) {
     @Override
     public boolean cellsCleanedUp(Transaction t, Set<Cell> cells) {
         SnapshotsStreamMetadataTable metaTable = tables.getSnapshotsStreamMetadataTable(t);
-        Collection<SnapshotsStreamMetadataTable.SnapshotsStreamMetadataRow> rows = Lists.newArrayListWithCapacity(cells.size());
+        Set<SnapshotsStreamMetadataTable.SnapshotsStreamMetadataRow> rows = Sets.newHashSetWithExpectedSize(cells.size());
         for (Cell cell : cells) {
             rows.add(SnapshotsStreamMetadataTable.SnapshotsStreamMetadataRow.BYTES_HYDRATOR.hydrateFromBytes(cell.getRowName()));
         }
+        SnapshotsStreamIdxTable indexTable = tables.getSnapshotsStreamIdxTable(t);
+        Set<SnapshotsStreamIdxTable.SnapshotsStreamIdxRow> indexRows = rows.stream()
+                .map(SnapshotsStreamMetadataTable.SnapshotsStreamMetadataRow::getId)
+                .map(SnapshotsStreamIdxTable.SnapshotsStreamIdxRow::of)
+                .collect(Collectors.toSet());
+        Map<SnapshotsStreamIdxTable.SnapshotsStreamIdxRow, Iterator<SnapshotsStreamIdxTable.SnapshotsStreamIdxColumnValue>> indexIterator
+                = indexTable.getRowsColumnRangeIterator(indexRows,
+                        BatchColumnRangeSelection.create(PtBytes.EMPTY_BYTE_ARRAY, PtBytes.EMPTY_BYTE_ARRAY, 1));
+        Set<SnapshotsStreamMetadataTable.SnapshotsStreamMetadataRow> rowsWithNoIndexEntries
```

Comment:
commenting here because it's easier to read,

if I've understood correctly, index table keeps track of references in dynamic columns
and you're just checking that for a given stream, its index row has no columns (references)?

perhaps rename `indexIterator` to `referenceIteratorByStream` and

`rowsWithNoIndexEntries` to `streamsWithNoReferences`

Reply:
Correct, that's what the code is doing (and what you describe would be more descriptive at the application level of what is going on - will change to those)